src/Http/Http.Abstractions/src/Extensions/MapMiddleware.cs

```diff
@@ -48,25 +48,29 @@ public async Task Invoke(HttpContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
-            PathString matchedPath;
-            PathString remainingPath;
-
-            if (context.Request.Path.StartsWithSegments(_options.PathMatch, out matchedPath, out remainingPath))
+            if (context.Request.Path.StartsWithSegments(_options.PathMatch, out PathString matchedPath, out PathString remainingPath))
             {
-                // Update the path
-                var path = context.Request.Path;
-                var pathBase = context.Request.PathBase;
-                context.Request.PathBase = pathBase.Add(matchedPath);
-                context.Request.Path = remainingPath;
+                PathString path = context.Request.Path;
```

Comment:
var is in our style guide. https://github.com/aspnet/AspNetCore/wiki/Engineering-guidelines#usage-of-the-var-keyword
```suggestion
                var path = context.Request.Path;
```